### DISCUSSION TITLE
Mark `XCTFail` with `@_disfavoredOverload`

### DISCUSSION THREAD

**Discussion by davdroman** [pointfreeco/swift-issue-reporting#45] [Ideas]
In situations where `XCTDynamicOverlay` is implicitly imported in tests (such as by importing `Dependencies`), it might be useful to mark `XCTFail` with `@_disfavoredOverload` so XCTest's `XCTFail` can take precedence in tests without ambiguities.

Any thoughts? Is it worth a PR?

**Comment by lukeredpath** [comment]
I'm running into this issue now in some test utilities that import XCTest and ComposableArchitecture - any ideas on how to disambiguate this? `XCTest.XCTFail` doesn't work. This started happening when updating from 0.45 to 0.49

**Comment by tgrapperon** [comment]
Cross-linking https://github.com/pointfreeco/swift-dependencies/issues/13 for future reference.

**Comment by stephencelis** [comment]
@davdroman Any interest in PRing this, or should we dive in? :smile:

**Reply by davdroman** [reply]
Sure! I'll put something together shortly.

**Reply by davdroman** [reply]
It's up 😄  https://github.com/pointfreeco/xctest-dynamic-overlay/pull/46
